src/web-auth/popup.js

```diff
@@ -187,50 +189,12 @@ Popup.prototype.authorize = function(options, cb) {
  * @param {credentialsCallback} cb
  */
 Popup.prototype.loginWithCredentials = function(options, cb) {
-  var params;
-  var popup;
-  var url;
-  var relayUrl;
-
-  /* eslint-disable */
-  assert.check(
-    options,
-    { type: 'object', message: 'options parameter is not valid' },
-    {
-      clientID: { optional: true, type: 'string', message: 'clientID option is required' },
-      redirectUri: { optional: true, type: 'string', message: 'redirectUri option is required' },
-      responseType: { optional: true, type: 'string', message: 'responseType option is required' },
-      scope: { optional: true, type: 'string', message: 'scope option is required' },
-      audience: { optional: true, type: 'string', message: 'audience option is required' }
-    }
-  );
-  /* eslint-enable */
-
-  popup = this.getPopupHandler(options);
-
-  options = objectHelper
-    .merge(this.baseOptions, [
-      'clientID',
-      'scope',
-      'domain',
-      'audience',
-      '_csrf',
-      'state',
-      '_intstate',
-      'nonce'
-    ])
-    .with(objectHelper.blacklist(options, ['popupHandler']));
-
-  params = objectHelper.pick(options, ['clientID', 'domain']);
-  params.options = objectHelper.toSnakeCase(
-    objectHelper.pick(options, ['password', 'connection', 'state', 'scope', '_csrf', 'device'])
-  );
-  params.options.username = options.username || options.email;
-
-  url = urljoin(this.baseOptions.rootUrl, 'sso_dbconnection_popup', options.clientID);
-  relayUrl = urljoin(this.baseOptions.rootUrl, 'relay.html');
-
-  return popup.load(url, relayUrl, { params: params }, responseHandler(cb));
+  options.realm = options.realm || options.connection;
+  delete options.connection;
```

Comment:
Is this intended to be used on legacy mode as well? If that's true, we should keep it.

Reply:
since we don't know if the user has oidc turned on or not, do you think we should send both?

Reply:
Probably legacy mode won't understand the `realm` parameter, and of course for the new one we shouldn't be sending deprecated parameters. Do we have other option than sending both? Can you please try if that works fine on both pipelines?